doc/devel/documenting_mpl.rst

```diff
@@ -29,349 +29,170 @@ requirements that are needed to build the documentation.  They are listed in
+             label='', **kwargs):
+        """
+        Plot horizontal lines at each *y* from *xmin* to *xmax*.
 
-  yields:
+        Parameters
+        ----------
+        y : scalar or sequence of scalar
+            y-indexes where to plot the lines.
 
-  .. deprecated:: 0.98
-     This feature is obsolete, use something else.
+        xmin, xmax : scalar or 1D array_like
+            Respective beginning and end of each line. If scalars are
+            provided, all lines will have same length.
 
-* Use the *versionadded* and *versionchanged* directives, which have similar
-  syntax to the *deprecated* role:
+        colors : array_like of colors, optional, default: 'k'
 
-  .. code-block:: rst
+        linestyles : ['solid' | 'dashed' | 'dashdot' | 'dotted'], optional
 
-    .. versionadded:: 0.98
-       The transforms have been completely revamped.
+        label : string, optional, default: ''
 
-  .. versionadded:: 0.98
-     The transforms have been completely revamped.
+        Returns
+        -------
+        lines : `~matplotlib.collections.LineCollection`
 
-* Use the *seealso* directive, for example:
+        Other Parameters
+        ----------------
+        **kwargs : `~matplotlib.collections.LineCollection` properties.
 
-  .. code-block:: rst
+        See also
+        --------
+        vlines : vertical lines
+        axhline: horizontal line across the axes
+        """
 
-    .. seealso::
+The Sphinx website also contains plenty of documentation_ concerning ReST
+markup and working with Sphinx in general.
 
-       Using ReST :ref:`emacs-helpers`:
-          One example
+Function arguments
+------------------
+Function arguments and keywords within docstrings should be referred to using
+the ``*emphasis*`` role. This will keep Matplotlib's documentation consistent
+with Python's documentation:
 
-       A bit about :ref:`referring-to-mpl-docs`:
-          One more
+.. code-block:: rst
 
-  yields:
+  Here is a description of *argument*
 
-  .. seealso::
+Please do not use the ```default role```:
 
-     Using ResT :ref:`emacs-helpers`:
-        One example
 
-     A bit about :ref:`referring-to-mpl-docs`:
-        One more
+.. code-block:: rst
 
-* Please keep the :ref:`glossary` in mind when writing documentation. You can
-  create a references to a term in the glossary with the ``:term:`` role.
+   Please do not describe `argument` like this.
 
-* The autodoc extension will handle index entries for the API, but additional
-  entries in the index_ need to be explicitly added.
+nor the ````literal```` role:
 
-* Keyword arguments should be described using a definition list.
+.. code-block:: rst
 
-  .. note::
-     Matplotlib makes extensive use of keyword arguments as pass-through
-     arguments, there are a many cases where a table is used in place of a
-     definition list for autogenerated sections of docstrings.
+   Please do not describe ``argument`` like this.
 
 Figures
 =======
 
```

Comment:
This could use an intoroductory paragraph, or even a change of heading.  I find it very mysterious what figures you are talking about.  In docstrings?  I think you mean in examples and tutorials.  If so, maybe a "Writing an example or tutorial" would be a more useful heading, and figure creation is of course the most important thing that happens in that context.

Reply:
Added a bit more text and changed the title; is it better now?

Reply:
I think maybe a specific section on writing an example/tutorial is needed.

Reply:
Maybe a new document? This one is quite long already, so sticking to just "Documenting code" is better, and a separate "Writing examples and tutorials" guide can be written.